src/Exceptions.jl

```diff
@@ -0,0 +1,14 @@
+"""
+Custom exception types used throughout Celeste.
+"""
+module Exceptions
+
+"""
+Some parameter to a function has invalid values. The message should explain what parameter is
+invalid and why.
+"""
+type InvalidInputError <: Exception
```

Comment:
Could we just define this type around where it's thrown, rather than creating a new file? A new module with just one type seems like a bit of a heavy weight solution.